flytekit/deck/deck.py

```diff
@@ -174,3 +174,29 @@ def get_deck_template() -> "Template":
         autoescape=select_autoescape(enabled_extensions=("html",)),
     )
     return env.get_template("template.html")
+
+
+class PythonDependencyDeck(Deck):
+    """
+    PythonDependencyDeck is a deck that contains information about the library name and version.
+    """
+
+    def __init__(self, name: Optional[str] = "Python Dependency", html: Optional[str] = ""):
+        super().__init__(name, html)
+
+    @property
+    def html(self) -> str:
+        import json
+        import subprocess
+
+        import pandas as pd
+        from flytekitplugins.deck.renderer import TableRenderer
+
+        try:
+            installed_packages = json.loads(subprocess.check_output(["pip", "list", "--format", "json"]))
+        except subprocess.CalledProcessError as e:
+            logger.error(f"Error occurred while fetching installed packages: {e}")
+            return ""
+        df = pd.DataFrame(installed_packages)
+        html = TableRenderer().to_html(df)
```

Comment:
Let's make this html a tad bit more complicated so that we can also solve the reproduction use-case (i.e. copy list of dependencies as a requirements.txt file), by doing something like:

```suggestion
        table = TableRenderer().to_html(df)
        html = f"""
        <!DOCTYPE html>
        <html lang="en">
        <head>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>Flyte Dependencies</title>
        <script>
        async function copyTable() {
        var table = document.getElementsByTagName('table')[0];
        var requirements_txt = '';

        for (var i = 0, row; row = table.rows[i]; i++) {
            if (i == 0) {
            continue;
            }
            requirements_txt += row.cells[0].innerText+""+row.cells[1].innerText+'\n';
        }

        try {
            await navigator.clipboard.writeText(requirements_txt);
        } catch (err) {
            console.log('Oops, unable to copy');
        }
        }
        </script>
        </head>
        <body>

        <svg width="100px" height="100px" viewBox="0 0 24 24" onclick="copyTable()" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
            <g stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
                <g fill="#212121" fill-rule="nonzero">
                    <path d="M5.50280381,4.62704038 L5.5,6.75 L5.5,17.2542087 C5.5,19.0491342 6.95507456,20.5042087 8.75,20.5042087 L17.3662868,20.5044622 C17.057338,21.3782241 16.2239751,22.0042087 15.2444057,22.0042087 L8.75,22.0042087 C6.12664744,22.0042087 4,19.8775613 4,17.2542087 L4,6.75 C4,5.76928848 4.62744523,4.93512464 5.50280381,4.62704038 Z M17.75,2 C18.9926407,2 20,3.00735931 20,4.25 L20,17.25 C20,18.4926407 18.9926407,19.5 17.75,19.5 L8.75,19.5 C7.50735931,19.5 6.5,18.4926407 6.5,17.25 L6.5,4.25 C6.5,3.00735931 7.50735931,2 8.75,2 L17.75,2 Z M17.75,3.5 L8.75,3.5 C8.33578644,3.5 8,3.83578644 8,4.25 L8,17.25 C8,17.6642136 8.33578644,18 8.75,18 L17.75,18 C18.1642136,18 18.5,17.6642136 18.5,17.25 L18.5,4.25 C18.5,3.83578644 18.1642136,3.5 17.75,3.5 Z">

        </path>
        {table}

        </body>
        </html>
        """
```


This renders like:
<img width="387" alt="Screenshot 2024-03-13 at 10 11 21 PM" src="https://github.com/flyteorg/flytekit/assets/653394/6618343c-5603-4b24-9819-12bddcd9416d">

The idea being that if you click on the icon you copy the list of requirements in the format that `requirements.txt` expects. Obviously someone who knows a lot more of html than me can get this to a better state (maybe just turn that into a button that says "Copy in requirements.txt format"?).

Reply:
> (maybe just turn that into a button that says "Copy in requirements.txt format"?).

I like this option more. The button is a bit too abstract for me.

Reply:
Not amazing, but if we swap the `<svg>` tag with:
```
<button onclick="copyTable()">
  <span>Copy table as requirements.txt</span>
  </button>
```

then this renders like:
<img width="392" alt="Screenshot 2024-03-14 at 8 47 17 AM" src="https://github.com/flyteorg/flytekit/assets/653394/a87ff481-8484-4839-927c-a835ed282c79">

Reply:
That is great! My only nit is to have some margin under the button.

As for implementation, I prefer to not parse the HTML table and include the raw `pip freeze` in the HTML and hide it. Then, the button will just need to read the text. (Also less javascript code for us to maintain)

Reply:
I put the new version of the html template in https://gist.github.com/eapolinario/dc45c349a582d287c1b2a77ac83a7fce

This assumes that there's a python variable called `requirements_txt` that will be passed to this string.

Reply:
updated

Reply:
I found that when using iframe, `iframe` needs to add the allow attribute
`allow="clipboard-write"`, so that the copy can be successful only after clicking the button.

Reply:
Results pasted into txt file, file content:
> adlfs==2024.2.0
> aiobotocore==2.12.1
> aiohttp==3.9.3
> aioitertools==0.11.0
> aiosignal==1.3.1
> ...
> ...
> ...
> wrapt==1.16.0
> yarl==1.9.4
> ydata-profiling==4.6.5
> zipp==3.18.1

Reply:
PR for adding attributes to iframe: https://github.com/flyteorg/flyteconsole/pull/852